core/app/models/concerns/spree/soft_deletable.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'discard'
+
+module Spree
+  module SoftDeletable
+    extend ActiveSupport::Concern
+
+    included do
+      include Discard::Model
+      self.discard_column = :deleted_at
+
+      default_scope { kept }
```

Comment:
`default_scope` has a bit of bad rap, see for example https://rails-bestpractices.com/posts/2013/06/15/default_scope-is-evil/

I'm not implying we should not use it at all, but we should make store maintainers explicitly aware of its existence, with a message such as please verify your discard models overrides don't include a `default_scope` and that the new `default_scope` does not interact weirdly with your custom code.

Reply:
Yeah, I understand the desire to use it here as it makes the migration significantly easier. I'd prefer we not use it by default for any future models we make soft-deletable though.